src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/MlInformation.cs

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Tizen.MachineLearning.Inference
+{
+    /// <summary>
+    /// The MlInformation class manages information based on key-value.
+    /// </summary>
+    /// <since_tizen> 12 </since_tizen>
+    public class MlInformation
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private InfoType _type;
+
+        private enum InfoType
+        {
+            Option = 0,
+            Information = 1,
+        }
+
+        /// <summary>
+        /// Creates a MlInformation instance.
+        /// </summary>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public MlInformation() {
+            NNStreamer.CheckNNStreamerSupport();
+
+            NNStreamerError ret = Interop.Util.CreateOption(out _handle);
+            NNStreamer.CheckException(ret, "Failed to create information handle");
+
+            _type = InfoType.Option;
+        }
+
+        /// <summary>
+        /// Creates a MlInformation instance from Native handle.
+        /// </summary>
+        /// <since_tizen> 12 </since_tizen>
+        internal MlInformation(IntPtr handle) {
```

Comment:
recommend adding param type for future work :)
```
internal MlInformation(IntPtr handle, InfoType type, bool fromList)
```

Reply:
Updated :)